data_structures/singly_linked_list.py

```diff
@@ -51,11 +43,11 @@ def delete_ele(self, val):
       raise SinglyLinkedListError("Empty List! No elements found.")   
     while p.value == val:
       p = p.succ
-    self.head = p 
+    self.head = p
     q = p
     p = p.succ
     while p is not None:
-      while p.value == val and p is not None:
```

Comment:
I know this one. It is called circuiting or something else. Which means that, the first condition in this while loop will be checked first. If it is wrong, the second one will no longer need to be checked. Thus, according to the relations between these two conditions, we should check if the node is none or not first.